src/Broadway/EventStore/InMemoryEventStore.php

```diff
@@ -53,6 +56,7 @@ public function append($id, DomainEventStreamInterface $eventStream)
             $this->assertPlayhead($this->events[$id], $playhead);
 
             $this->events[$id][$playhead] = $event;
+            $this->allEvents[] = $event;
```

Comment:
### Ordering guarantees

Ah! Check. So one thing we haven't discussed so far is the guarantees we want to provide in `visitEvents()`. I think we should:

1. _guarantee_ events per aggregate are replayed in the order they were saved
2. _encourage_ chronical ordering

For the SQL world providing both (1) and (2) makes sense, but imagine an event store using a document based model. One approach there would be to store one document per id, containing all the events. While guaranteeing (1) is easy to do, guaranteeing (2) will be hard.

### In Memory

In case of the in memory implementation it is (probably?) only used in tests. Most tests focus on one aggregate root id anyway. While we encourage total chronological ordering, I think it might actually be a good idea to not provide that ordering in the in memory event store, so that people will be made aware of the guarantees we make.